### ISSUE TITLE
[11.0] hr_timesheet_attendance - attendances in report not adjusted for timezone

### ISSUE THREAD

**Issue by bennals** [odoo/odoo#37104]
Steps to reproduce:

- Change a users timezone to Australia/Brisbane (UTC +10).

- Create an attendance record with check-in at 7.30am on 19/9/2019 and check-out at 9.30am on 19/9/2019

- Click on "Timesheets / Attendance" report  in the timesheets app.

Expected result

There are 2 hours in the "Total attendance" column for **19** Sep 2019.

Actual result

There are 2 hours in the "Total attendance" column for **18** Sept 2019.

It seems that the report is using the UTC time (9.30pm on 18 Sept 2019 - Brisbane time minus 10 hours) to enter in the pivot report instead of the timezone adjusted datetime.

This results in a confusing display, but will also create a discrepancy between the attendance and timesheet records for users based in timezones where the time difference pushes check-ins into the previous day.

Impacted versions:  11.0 CE

![image](https://user-images.githubusercontent.com/7786180/65195165-69cb8680-dac1-11e9-9214-7fd5ae525840.png)

**Comment by pedrobaeza** [comment]
Is this still happening with latest code?

**Comment by bennals** [comment]
Yes it is @pedrobaeza .  Just checked in the latest V11 on runbot.

![Screen Shot 2019-11-04 at 9 52 54 PM](https://user-images.githubusercontent.com/7786180/68119293-0f419900-ff4e-11e9-8dbc-7ccfaa46f0de.png)
![Screen Shot 2019-11-04 at 9 54 48 PM](https://user-images.githubusercontent.com/7786180/68119294-0f419900-ff4e-11e9-9bf9-623de17e7b94.png)

**Comment by pedrobaeza** [comment]
OK, maybe @RomainLibert can give a check to this.

**Comment by bennals** [comment]
@RomainLibert just giving this a nudge before I try and manufacture a clunky work around with Google Spreadsheet:)  Still happening on latest runbot.

Would be nice if it got some attention prior to 11 becoming end-of-support.  Crossing fingers.

**Comment by thomich** [comment]
Hey

This must be reopened.. cause it's still a Problem in v17.

It also affect 14, 15.. maybe also 16.

Tested on runbot: https://63958056-17-0-all.runbot135.odoo.com/

Timezone of user set to Europe/Zurich

Attendance at 00:39
![image](https://github.com/odoo/odoo/assets/4398840/f04663a9-0a3d-47f4-894a-cae3679d2414)

Timesheet/Attendance report, nothing on 13.6.
![image](https://github.com/odoo/odoo/assets/4398840/617c2ae4-0cac-4068-a1d5-991bb58f8da4)


If change attendance start to 02:39, report looks like:
![image](https://github.com/odoo/odoo/assets/4398840/4d7a41c0-9cad-4823-b073-7f535acadc50)

Fix for all versions is welcome...

**Comment by bennals** [comment]
Good luck Tomich, though I think you will do better creating a new issue.  From memory I checked the V12 runbot at the time and the issue seemed fixed, so I'm surprised to hear it reappeared in later versions.

**Comment by thomich** [comment]
this are the only diff's i found between 12 and 14... nothing todo about user timezone

[hr_timesheet_attendance_report_view.txt](https://github.com/user-attachments/files/15818534/hr_timesheet_attendance_report_view.txt)
[hr_timesheet_attendance_report.txt](https://github.com/user-attachments/files/15818535/hr_timesheet_attendance_report.txt)

why new issue, reopen and change focus on version...

**Comment by bennals** [comment]
ok...I'm not a developer so I didn't delve into the code - just checked runbot, but it was five years ago and maybe my quick test was flawed.

My suggestion to open a fresh issue is only based on the amount of attention my original issue received (ie, not much!).  A new one , that isn't 'closed', five years old, and has a '17.0' in front of it, might catch someone's attention more?  Plus that was the suggestion from C3POdoo (yes, automated, but still...) further up the thread.

**Comment by thomich** [comment]
https://github.com/odoo/odoo/issues/169592
